Source/WebCore/accessibility/cocoa/AXTextMarkerCocoa.mm

```diff
@@ -87,8 +90,34 @@
         // FIXME: Should we be passing SpellCheck::Yes? Maybe we need to take `SpellCheck` as a function parameter?
         return start.isolatedObject()->createAttributedString(start.runs()->substring(minOffset, maxOffset - minOffset), AXCoreObject::SpellCheck::No).autorelease();
     }
-    // FIXME: Handle ranges that span multiple objects.
-    return nil;
+
+    RetainPtr<NSMutableAttributedString> result = start.isolatedObject()->createAttributedString(start.runs()->substring(start.offset()), AXCoreObject::SpellCheck::No);
+    auto emitNewlineOnExit = [&] (AXIsolatedObject& object) {
+        // FIXME: This function should not just be emitting newlines, but instead handling every character type in TextEmissionBehavior.
+        auto behavior = object.emitTextAfter();
+        if (behavior != TextEmissionBehavior::Newline && behavior != TextEmissionBehavior::DoubleNewline)
+            return;
+
+        auto length = [result length];
+        // Like TextIterator, don't emit a newline if the most recently emitted character was already a newline.
+        if (length && [[result string] characterAtIndex:length - 1] != '\n') {
+            // FIXME: This is super inefficient. We are creating a whole new dictionary and attributed string just to append newline(s).
+            NSString *newlineString = behavior == TextEmissionBehavior::Newline ? @"\n" : @"\n\n";
+            NSDictionary *attributes = [result attributesAtIndex:length - 1 effectiveRange:nil];
+            [result appendAttributedString:adoptNS([[NSAttributedString alloc] initWithString:newlineString attributes:attributes]).get()];
+        }
+    };
+
+    // FIXME: If we've been given reversed markers, i.e. the end marker actually comes before the start marker,
+    // we may want to detect this and try searching AXDirection::Previous?
+    RefPtr current = findObjectWithRuns(*start.isolatedObject(), AXDirection::Next, std::nullopt, emitNewlineOnExit);
+    while (current && current->objectID() != end.objectID()) {
+        [result appendAttributedString:current->createAttributedString(current->textRuns()->toString(), AXCoreObject::SpellCheck::No).get()];
+        current = findObjectWithRuns(*current, AXDirection::Next, std::nullopt, emitNewlineOnExit);
+    }
+    [result appendAttributedString:end.isolatedObject()->createAttributedString(end.runs()->substring(0, end.offset()), AXCoreObject::SpellCheck::No).get()];
```

Comment:
Does the result of createAttributedString() need to be autoreleased?

Reply:
For these temporary strings, I think so — good call! I added that in the latest commit. Let me know if you think I'm missing any else.